src/socket.rs

```diff
@@ -1615,6 +1615,101 @@ impl Socket {
                 .map(|recv_tos| recv_tos > 0)
         }
     }
+
+    /// Set value for `IP_DONTFRAG` option of this socket
+    #[cfg(not(any(
+        target_os = "netbsd",
+        target_os = "openbsd",
+        target_os = "redox",
+        target_os = "vita",
+        target_os = "dragonfly",
+        target_os = "fuchsia",
+    )))]
```

Comment:
Can you do the inverse of this, that is specifying why OS do support this option.

Reply:
I don't know how other OS work. I changed the cfg to macos/ios/linux/windows. Other platforms need others to fill in